src/components/collection/Items.tsx

```diff
@@ -0,0 +1,162 @@
+import { QueryClientContext } from '../QueryClientContext';
+import { FileChildrenCard, FolderChildrenCard } from './ChildrenCard';
+
+const DEFAULT_ITEM_SHOWN_COUNT = {
+  xs: 4,
+  sm: 4,
+  md: 6,
+  lg: 6,
+  xl: 8,
+};
+
+type CollapsibleItemCategoryProps = {
+  items: Immutable.List<ItemRecord>;
+  defaultItemCount: number;
+  children: (item: ItemRecord) => JSX.Element;
+};
+
+const CollapsibleItemCategory: React.FC<CollapsibleItemCategoryProps> = ({
+  items,
+  children,
+  defaultItemCount,
+}) => {
+  const { t } = useTranslation();
+
+  const [showMoreItems, setShowMoreItems] = useState<boolean>(false);
+
+  const shownItems = React.useMemo(() => {
+    if (showMoreItems) {
+      return items;
+    }
+    return items.slice(0, defaultItemCount);
+  }, [items, showMoreItems, defaultItemCount]);
+
+  const handleShowMoreItems = () => {
+    setShowMoreItems((prevValue) => !prevValue);
+  };
+
+  const additionalItemsCount = items.size - defaultItemCount;
+
+  return (
+    <>
+      <Box display="flex" justifyContent="flex-end">
+        {items.size > defaultItemCount && (
+          <Button
+            onClick={handleShowMoreItems}
+            startIcon={showMoreItems ? <Remove /> : <Add />}
+          >
+            {showMoreItems
+              ? t(LIBRARY.SUMMARY_ITEMS_SHOW_LESS, {
+                  count: additionalItemsCount,
+                })
+              : t(LIBRARY.SUMMARY_ITEMS_SHOW_MORE, {
+                  count: additionalItemsCount,
+                })}
+          </Button>
+        )}
+      </Box>
+      <Grid container spacing={2} id={CHILDREN_ITEMS_GRID_ID} marginBottom={3}>
+        {shownItems.map((item) => (
+          <Grow in key={item.id}>
+            <Grid item xs={12} sm={12} md={6} lg={4} xl={3}>
+              {children(item)}
+            </Grid>
+          </Grow>
+        ))}
+      </Grid>
+    </>
+  );
+};
+
+type ItemsProps = {
+  parentId: string;
+  lang: string | undefined;
+  isTopLevel: boolean;
+};
+
+const Items: React.FC<ItemsProps> = ({ parentId, lang, isTopLevel }) => {
+  const { t } = useTranslation();
+  const { hooks } = useContext(QueryClientContext);
+  const { data: items } = hooks.useChildren(parentId, {
+    placeholderData: List(PLACEHOLDER_COLLECTION.children),
+  });
+
+  const theme = useTheme();
+
+  const extraSmall = useMediaQuery(theme.breakpoints.down('sm'));
+  const small = useMediaQuery(theme.breakpoints.down('md'));
+  const medium = useMediaQuery(theme.breakpoints.down('lg'));
+  const large = useMediaQuery(theme.breakpoints.down('xl'));
+
+  const itemToShow = React.useMemo(() => {
+    if (extraSmall) {
+      return DEFAULT_ITEM_SHOWN_COUNT.xs;
+    }
+    if (small) {
+      return DEFAULT_ITEM_SHOWN_COUNT.sm;
+    }
+    if (medium) {
+      return DEFAULT_ITEM_SHOWN_COUNT.md;
+    }
+    if (large) {
+      return DEFAULT_ITEM_SHOWN_COUNT.lg;
+    }
+    return DEFAULT_ITEM_SHOWN_COUNT.xl;
+  }, [extraSmall, small, medium, large]);
```

Comment:
My comment may not be correct because V8 may optimize this away anyway, but wouldn't it be technically more efficient to inline the `useMediaQuery` calls inside the if conditions? If a smaller condition is met, the other calls would be short-circuited. @spaenleh

Reply:
> Wouldn't that violate the rule of hooks ? since they would be called conditionally ?

Makes sense, I did not capture that they were hooks as well